source/common/common/win32/thread_impl.cc

```diff
@@ -0,0 +1,31 @@
+#include <process.h>
+
+#include "common/common/assert.h"
+#include "common/common/thread_impl.h"
+
+namespace Envoy {
+namespace Thread {
+
+ThreadImpl::ThreadImpl(std::function<void()> thread_routine) : thread_routine_(thread_routine) {
```

Comment:
It looks like this is the exact same name as the posix one, so the decision is made at link-time, and if you link both files in you'd get a duplicate symbol conflict.

That's not really a concern for windows vs posix, but you may also want an alternate thread system even if more than one is possible on the platform. That's something that we need (google has some custom thread infrastructure).

I'll give you another example: you may want a stats-collecting thread-system decorator that keeps a gauge of the active thread-count or other metrics.  I'd keep that separate from the platform-specific impls, so (e.g.) you could use that to decorate either windows or posix one.

Can we inject the right platform-specificThreadFactory where needed rather than having this be a link-time decision?